### ISSUE TITLE
Issue with cleanup in TCP connection, related to rtx_timer

### ISSUE THREAD

**Issue by fwsGonzo** [hioa-cs/IncludeOS#610]
Crash after rtx_timeout on any connection

**Comment by fwsGonzo** [comment]
This needs to be reproduced now that we have fixed many network issues
I will close this if nothing happens for a week

**Comment by fwsGonzo** [comment]
```
[2] 0x3 + 0x003: (null) [3] 0x217568 + 0x028: net::TCP::Connection::retransmit() [4] 0x21792e + 0x07e: net::TCP::Connection::rtx_timeout() [5] 0x28b1a2 + 0x012: void delegate<void ()>::method_stub<net::TCP::Connection, &(net::TCP::Connection::rtx_timeout())>(void*) [6] 0x20c171 + 0x141: hw::PIT::irq_handler()
```
Received on my IRC server. Probably not too hard to replicate, given that we create some test that forces the rtx timers to active en-masse.

**Comment by AndreasAakesson** [comment]
Possibly fixed with #700?

**Comment by fwsGonzo** [comment]
Yes, this was the last fix
